helm/cluster-azure/values.schema.json

```diff
@@ -233,6 +241,38 @@
                     "default": "1.24.11",
                     "title": "Kubernetes version",
                     "type": "string"
+                },
+                "network": {
```

Comment:
are these settings

```
    internal:
      privateDNSZoneName: np2235.azuretest.gigantic.io
      network:
        vpn:
          gatewayMode: remote
```

supposed to be specified for every cluster ? 

in that case they will need to be moved to the `connectivity` section or they won't be exposed in the UI 

Can we not generate all of those from `connectivity.network.mode == private -> set all those values ?`

Reply:
> supposed to be specified for every cluster ?

`privateDNSZoneName` can be specific for every cluster indeed, just still not sure if all WCs in the same MC must have the same "base domain". In that case here we could have this auto-generated (if "base domain" is available in the config).

`gatewayMode: remote` is set for workload clusters. If, for whatever reason, some peering is added on the MC itself (AzureCluster CR of the MC), then it must be set to `gatewayMode: local`, because the VPN gateway is deployed in the MC.

Basically:
- Cluster that is using VPN gateway from (remote) peered VNet has `gatewayMode: remote`. Currently these are all workload clusters.
- Cluster that is using VPN gateway from its own (local) VNet has `gatewayMode: local`. Currently this are the management clusters.

When peering 2 VNets `A` and `B` there are 2 peering that get created, `A -> B` and `B -> A` and the above difference exist because properties of those peerings are created with slightly different properties, depending on which VNet has VPN gateway deployed in it.

I'll try to document this in a better way (if we had Azure VPN as a part of the AzureCluster CR then it would be easier to automatically set these fields to correct values, because you could know where the VPN is deployed, but now we have to set it explicitly).

> they won't be exposed in the UI

Wasn't aware of that. What UI specifically, you mean Happa?

I've put in internal as I'm not really sure how this should look like in the long run, so putting it in the `internal` makes it less exposed (as intended).

In any case good shoutout, I'll see how it can be made nicer.

Reply:
> Can we not generate all of those from connectivity.network.mode == private -> set all those values ?

Bottom line, not really, or at least not that easy or possible at the moment, especially since there is a difference how something is done in the MC vs in the WC, since we use `cluster-azure` app for both, and we want that these `cluster-xxx` apps are fully agnostic when it comes to knowing if you are deploying MC or WC, so in order to accomplish that we have to be more explicit.

Reply:
Updated the change so `privateDNSZoneName` in the CR now uses `baseDomain` from config.

Reply:
So now `privateDNSZoneName` doesn't have to be specified, and `gatewayMode: remote` is the default anyway  (because that works for workload clusters and it is overridden only for MCs) so also that does not have to be specified explicitly.